ios/StripeTerminalReactNative.swift

```diff
@@ -1003,13 +995,26 @@ class StripeTerminalReactNative: RCTEventEmitter, DiscoveryDelegate, BluetoothRe
     }
 
     func terminal(_ terminal: Terminal, didChange offlineStatus: OfflineStatus) {
-        let offlineStatus = Mappers.mapFromOfflineStatus(offlineStatus)
-        sendEvent(withName: ReactNativeConstants.CHANGE_OFFLINE_STATUS.rawValue, body: ["result": offlineStatus])
+        let result = Mappers.mapFromOfflineStatus(offlineStatus)
+        sendEvent(withName: ReactNativeConstants.CHANGE_OFFLINE_STATUS.rawValue, body: ["result": result])
     }
 
     func terminal(_ terminal: Terminal, didForwardPaymentIntent intent: PaymentIntent, error: Error?) {
         let result = Mappers.mapFromPaymentIntent(intent, uuid: "")
-        sendEvent(withName: ReactNativeConstants.FORWARD_PAYMENT_INTENT.rawValue, body: ["result": result])
+        var body: [String: Any] = ["result": result]
+        
+        if let nsError = error as NSError? {
+           let errorAsDictionary = Errors.createError(nsError: nsError)
+            // createError will return a dictionary of ["error": {the error}]
+            // so merge that with the result so we have [result:, error:]
+            body = body.merging(errorAsDictionary, uniquingKeysWith: { _, error in
```

Comment:
getting a warning for this line, do people with more iOS know-how have tips?
`Expression implicitly coerced from '[String : Any?]' to '[String : Any]'`

Reply:
since `createError` returns an `[String: Any?]` if you type `body` as `[String: Any?]` it should be happy. Though it's not clear to me why `createError(nsError: NSError) -> [String: Any?] {` returns a `[String: Any?]`...from what I can tell it calls another variant that is [String: Any] and so the values should never be nil 🤔 

mind removing that ? from `createError(nsError: NSError) -> [String: Any?] {`. that should fix it and I think is what that method should be typed as

Reply:
@bric-stripe done